src/main.leo

```diff
@@ -229,50 +245,64 @@ program privx_exchange_beta4.aleo {
         Mapping::set(balance, to_key, to_current_balance);
     }
     
-    transition deposit(token_id: u64, amount: u64) {
-        return then finalize(self.caller, token_id, amount);
+    // Transfer from `default_balance_account` to `deposit_account`
+    transition deposit(token_id: u64, amount: u64, account: privx_account.leo/Account) {
```

Comment:
Besides depositing public balances we might also enable depositing private Token Record